tests/spark_autologging/test_pyspark_ml_autologging.py

```diff
@@ -0,0 +1,216 @@
+    mlflow.end_run()
+
+
+def test_meta_estimator_fit_performs_logging_only_once(dataset_binomial):
+    from pyspark.ml.classification import LogisticRegression, OneVsRest
+    mlflow.pyspark.ml.autolog()
+    with mock.patch("mlflow.log_params") as mock_log_params, \
+            mock.patch("mlflow.set_tags") as mock_set_tags, \
+            mock.patch("mlflow.spark.log_model") as mock_log_model:
+        with mlflow.start_run() as run:
+            lor = LogisticRegression()
+            ova = OneVsRest(classifier=lor)
+            ova.fit(dataset_binomial)
+
+        mock_log_params.assert_called_once()
+        mock_set_tags.assert_called_once()
+        mock_log_model.assert_called_once()
+
+        query = "tags.{} = '{}'".format(MLFLOW_PARENT_RUN_ID, run.info.run_id)
+        assert len(mlflow.search_runs([run.info.experiment_id])) == 1
+        assert len(mlflow.search_runs([run.info.experiment_id], query)) == 0
+
+
+def test_fit_with_params(dataset_binomial):
+    mlflow.pyspark.ml.autolog()
+    lr = LinearRegression()
+    extra_params = {lr.maxIter: 3, lr.standardization: False}
+    with mlflow.start_run() as run:
+        lr.fit(dataset_binomial, params=extra_params)
+    run_id = run.info.run_id
+    run_data = get_run_data(run_id)
+    assert run_data.params == truncate_param_dict(stringify_dict_values(
+        _get_instance_param_map(lr.copy(extra_params))))
+
+
+def test_fit_with_a_list_of_params(dataset_binomial):
+    mlflow.pyspark.ml.autolog()
+    lr = LinearRegression()
+    extra_params = {lr.maxIter: 3, lr.standardization: False}
+    extra_params2 = {lr.maxIter: 4, lr.standardization: True}
+    # Test calling fit with a list/tuple of paramMap
+    for params in [[extra_params, extra_params2], (extra_params, extra_params2)]:
+        with mock.patch("mlflow.log_params") as mock_log_params, \
+                mock.patch("mlflow.set_tags") as mock_set_tags:
+
+            with mlflow.start_run() as run:
+                with mock.patch("mlflow.pyspark.ml._logger.warning") as mock_warning:
+                    lr.fit(dataset_binomial, params=params)
+                    mock_warning.called_once_with(
+                        _get_warning_msg_for_fit_call_with_a_list_of_params(lr))
+            mock_log_params.assert_not_called()
+            mock_set_tags.assert_not_called()
+
+
+def test_should_log_model(dataset_binomial, dataset_multinomial):
+    from pyspark.ml.classification import LogisticRegression, OneVsRest
+    mlflow.pyspark.ml.autolog(log_models=True)
+    lor = LogisticRegression()
+
+    ova1 = OneVsRest(classifier=lor)
+    mlor_model = lor.fit(dataset_multinomial)
+    assert _should_log_model(mlor_model)
+
+    ova_model = ova1.fit(dataset_multinomial)
+    assert _should_log_model(ova_model)
+
+    with mock.patch("mlflow.pyspark.ml._log_model_allowlist",
+                    'pyspark.ml.regression.LinearRegressionModel\n'
+                    'pyspark.ml.classification.OneVsRestModel'):
```

Comment:
```suggestion
                   ['pyspark.ml.regression.LinearRegressionModel'
                    'pyspark.ml.classification.OneVsRestModel']):
```

The patched value should be a list, right?